programs/mango-v3-reimbursement/src/instructions/create_group.rs

```diff
@@ -0,0 +1,42 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::Token;
+
+use crate::state::Group;
+use crate::Error;
+
+#[derive(Accounts)]
+#[instruction(group_num: u32)]
+pub struct CreateGroup<'info> {
+    #[account(
+        init,
+        seeds = [b"Group".as_ref(), &group_num.to_le_bytes()],
+        bump,
+        payer = payer,
+        space = 8 + std::mem::size_of::<Group>(),
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    pub authority: Signer<'info>,
+
+    pub token_program: Program<'info, Token>,
+    pub system_program: Program<'info, System>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+pub fn handle_create_group(
+    ctx: Context<CreateGroup>,
+    group_num: u32,
+    table: Pubkey,
+    claim_transfer_destination: Pubkey,
+) -> Result<()> {
+    let mut group = ctx.accounts.group.load_init()?;
+    group.group_num = group_num;
+    group.table = table;
```

Comment:
For safety, I'd prefer if the table was passed as an account and that basic sanity checking was done here. Maybe the program prints out the total number of rows in the table, or the first row or something, just to show that it reads the data correctly.

We could even verify that the buffer account authority is the same as the group authority? (for non-testing groups)

Also, please log pubkeys passed to the instruction - that makes it easier to audit the create_group tx later because one doesn't need to manually unpack the data.